<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2025-12-15" />

<title>Using the PAC package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Using the PAC package</h1>
<h4 class="date">2025-12-15</h4>



<p>This vignette illustrates the basic usage of the PAC package for
R.</p>
<div id="biology-example" class="section level2">
<h2>Biology Example</h2>
<p>The PAC-MAN data analysis pipeline can be applied to mass-cytometry
(CyTOF) data analysis. In this case, the user reads in the example data
files (already saved as the Rdata format) subsetted from Bendall et al.,
2011 and goes through the data analysis pipeline.</p>
<p>Load the required R packages</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(PAC)</span></code></pre></div>
<p>Construct the sampleIDs vector to analyze the data</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>sampleIDs<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;BCR&quot;</span>, <span class="st">&quot;IL7&quot;</span>)</span></code></pre></div>
<p>Partition, cluster into desired number of subpopulations, and output
subpopulation mutual information networks</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">samplePass</span>(sampleIDs, <span class="at">dim_subset=</span><span class="cn">NULL</span>, <span class="at">hyperrectangles=</span><span class="dv">35</span>, <span class="at">num_PACSupop=</span><span class="dv">25</span>, <span class="at">num_networkEdge=</span><span class="dv">25</span>, <span class="at">max.iter=</span><span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Input Data: 2650 by 18
## Partition method: Discrepancy based partition
## Maximum level: 35
## partition completed
## [1] &quot;Initial Clustering...&quot;
## [1] &quot;Merging...&quot;</code></pre>
<pre><code>## Warning: `funs()` was deprecated in dplyr 0.8.0.
## ℹ Please use a list of either functions or lambdas:
## 
## # Simple named list: list(mean = mean, median = median)
## 
## # Auto named with `tibble::lst()`: tibble::lst(mean, median)
## 
## # Using lambdas list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
## ℹ The deprecated feature was likely used in the PAC package.
##   Please report the issue to the authors.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<pre><code>## Input Data: 3537 by 18
## Partition method: Discrepancy based partition
## Maximum level: 35
## partition completed
## [1] &quot;Initial Clustering...&quot;
## [1] &quot;Merging...&quot;</code></pre>
<pre><code>## Input Data: 3813 by 18
## Partition method: Discrepancy based partition
## Maximum level: 35
## partition completed
## [1] &quot;Initial Clustering...&quot;
## [1] &quot;Merging...&quot;</code></pre>
<p>Multiple Alignments of Networks</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>clades_network_only<span class="ot">&lt;-</span><span class="fu">MAN</span>(sampleIDs, <span class="at">num_PACSupop=</span><span class="dv">25</span>, <span class="at">smallSubpopCutoff=</span><span class="dv">100</span>, <span class="at">k_clades=</span><span class="dv">5</span>)</span></code></pre></div>
<p>Refine the PAC labels with multiple alignments of networks
representative labels for clades</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">refineSubpopulationLabels</span>(sampleIDs,<span class="at">dim_subset=</span><span class="cn">NULL</span>, clades_network_only, <span class="at">expressionGroupClamp=</span><span class="dv">5</span>)</span></code></pre></div>
<p>Draw clade/representative mutual information networks</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">getRepresentativeNetworks</span>(sampleIDs, <span class="at">dim_subset=</span><span class="cn">NULL</span>, <span class="at">SubpopSizeFilter=</span><span class="dv">200</span>, <span class="at">num_networkEdge=</span><span class="dv">25</span>)</span></code></pre></div>
<p>Obtain annotations of subpopulations</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>aggregateMatrix_withAnnotation<span class="ot">&lt;-</span><span class="fu">annotateClades</span>(sampleIDs, <span class="at">topHubs=</span><span class="dv">4</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">head</span>(aggregateMatrix_withAnnotation)</span></code></pre></div>
<pre><code>##   Annotation ClusterID SampleID   pPLCgamma2    pSTAT5      Ki67     pSHP2
## 1               clade1    Basal  0.868677090 1.4210433 1.6045171 1.0443365
## 2               clade2    Basal  0.417650152 0.7512015 0.7591787 0.6073356
## 3               clade3    Basal -0.009067766 0.1439336 0.5970280 0.0675446
## 4               clade1      BCR  1.414513278 1.4852767 1.6055084 1.3134441
## 5               clade2      BCR  0.331270112 0.7172607 0.2062699 0.5510241
## 6               clade4      BCR  0.912350993 1.5864291 3.5343868 0.9396143
##     pERK1.2 pMAPKAPK2 pZAP70.Syk    pSTAT3       pSLP     pNFkB IkBalpha
## 1 1.4661011 1.5628676 1.43355321 2.1289258 0.92427696 2.5118414 1.930353
## 2 0.9891308 0.9838982 0.68705268 1.4899395 0.46984991 1.7826376 1.426169
## 3 0.1806816 0.2359121 0.05943879 0.1938544 0.06218395 0.4461451 0.206577
## 4 1.5878684 1.6812649 1.74982011 2.3305502 1.16167123 2.5682406 1.835245
## 5 0.7372884 1.0448433 0.73709113 1.4776139 0.33423706 1.7574310 1.692790
## 6 1.1513465 1.7927573 1.14747844 2.1468960 0.69638449 2.4373202 1.627335
##         pH3      pP38  pBtk.Itk       pS6    pSrcFK      pCREB      pCrkL count
## 1 2.0931644 2.3369804 2.6054058 1.5696688 2.9274172 1.80653721 1.03088589  1481
## 2 0.9924652 1.5957836 1.7199279 0.6538710 1.8988397 0.86030693 0.57582728   507
## 3 0.2719665 0.2130849 0.5528641 0.1526701 0.2939679 0.08732748 0.06956507   458
## 4 2.6952983 2.8117867 2.5400931 1.8602256 3.1488323 1.91621956 1.04601166   990
## 5 1.2809057 1.7151455 1.9010833 0.9869888 2.2340015 1.26005371 0.47350443  1341
## 6 2.1106601 2.3534429 5.0717325 2.3080344 2.1823834 1.48173361 0.66764625   437</code></pre>
<p>Append subpopulation proportions for each sample in the annotation
matrix</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>annotationMatrix_prop<span class="ot">&lt;-</span><span class="fu">annotationMatrix_withSubpopProp</span>(aggregateMatrix_withAnnotation)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="fu">head</span>(annotationMatrix_prop)</span></code></pre></div>
<pre><code>##   Annotation ClusterID SampleID   pPLCgamma2    pSTAT5      Ki67     pSHP2
## 1               clade1    Basal  0.868677090 1.4210433 1.6045171 1.0443365
## 2               clade2    Basal  0.417650152 0.7512015 0.7591787 0.6073356
## 3               clade3    Basal -0.009067766 0.1439336 0.5970280 0.0675446
## 4               clade1      BCR  1.414513278 1.4852767 1.6055084 1.3134441
## 5               clade2      BCR  0.331270112 0.7172607 0.2062699 0.5510241
## 6               clade4      BCR  0.912350993 1.5864291 3.5343868 0.9396143
##     pERK1.2 pMAPKAPK2 pZAP70.Syk    pSTAT3       pSLP     pNFkB IkBalpha
## 1 1.4661011 1.5628676 1.43355321 2.1289258 0.92427696 2.5118414 1.930353
## 2 0.9891308 0.9838982 0.68705268 1.4899395 0.46984991 1.7826376 1.426169
## 3 0.1806816 0.2359121 0.05943879 0.1938544 0.06218395 0.4461451 0.206577
## 4 1.5878684 1.6812649 1.74982011 2.3305502 1.16167123 2.5682406 1.835245
## 5 0.7372884 1.0448433 0.73709113 1.4776139 0.33423706 1.7574310 1.692790
## 6 1.1513465 1.7927573 1.14747844 2.1468960 0.69638449 2.4373202 1.627335
##         pH3      pP38  pBtk.Itk       pS6    pSrcFK      pCREB      pCrkL count
## 1 2.0931644 2.3369804 2.6054058 1.5696688 2.9274172 1.80653721 1.03088589  1481
## 2 0.9924652 1.5957836 1.7199279 0.6538710 1.8988397 0.86030693 0.57582728   507
## 3 0.2719665 0.2130849 0.5528641 0.1526701 0.2939679 0.08732748 0.06956507   458
## 4 2.6952983 2.8117867 2.5400931 1.8602256 3.1488323 1.91621956 1.04601166   990
## 5 1.2809057 1.7151455 1.9010833 0.9869888 2.2340015 1.26005371 0.47350443  1341
## 6 2.1106601 2.3534429 5.0717325 2.3080344 2.1823834 1.48173361 0.66764625   437
##   subpop_proportion
## 1             60.55
## 2             20.73
## 3             18.72
## 4             29.43
## 5             39.86
## 6             12.99</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
